### DISCUSSION TITLE
Elephant's Foot File Correction

### DISCUSSION THREAD

**Discussion by JGarrett7** [sn4k3/UVtools#651] [Ideas]
I am wondering how easy it would be to add an elephant's foot correction. Obviously, most slicers already have this, but I am thinking of a correction type I haven't seen them employ before. Usually, elephant's foot is corrected by reducing the original shape's size to a smaller one by some offset value, but this requires you to figure out that offset.

I am wondering if there might be a more general solution. If one were to shine first the original image for the normal  (not first layers) amount of time, it wouldn't have elephant's foot, but it also wouldn't stick well. Is it possible to shine both that image and then an offset image on the same layer? I'm not sure how it would work in terms of the sliced file, but it could be implemented by either having two different layers, but no z height raise between them, or one layer where some pixels turn off partway through the layer. This would mean that you could set your offset value to anything more than the normal offset value as long as it still exposes enough surface for bed adhesion. The result of having an offset value higher than the one you need for the traditional method would just be a smaller high-exposure area.

I've attached an image that hopefully explains what I mean, but if not, let me know and I'll try again. I can't take credit for this idea. This guy thought of it first and he may explain it better than I've been able to https://www.youtube.com/watch?v=piz9VY-1Cfk&t=2s&ab_channel=Nerdtronic.
![ElephantsFoot](https://user-images.githubusercontent.com/43509109/214659715-7f1d601a-d2af-4d9f-95cd-a945b0e24cd2.jpg)

**Comment by sn4k3** [comment]
> but no z height raise between them

Not possible with most formats (Non gcode)

UVtools offer many methods for EF, one is called "Multiple exposure" which cures extended layer with an offset and then cures the rest at normal time, solving the dimension problem. However, you need a compatible printer.

See the tool below:

![image](https://user-images.githubusercontent.com/113281/214702832-61457094-f8da-45c7-b72a-a2faebae23fd.png)

Result (Camel is the first exposure): 
![image](https://user-images.githubusercontent.com/113281/214703102-4f9b4552-3b96-4ae8-a243-77e7582e13b3.png)

**Reply by JGarrett7** [reply]
Ah, well that sounds like exactly what I was looking for. Thanks!
